### PR TITLE
Hash Table in JavaScript

### PR DESCRIPTION
**Pull request by TidbitsJS**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Created a Hash Table in JavaScript, with the functionality of -
1. Insert item
2. Search item
3. Display items
4. Delete item

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#2137

### CODE REVIEW

**Comment by TidbitsJS** [comment]
Please review @HarshCasper

**Comment by HarshCasper** [comment]
Contributions for GSSoC '21 would be reviewed, after the Coding Period starts. Thanks!

File: JavaScript/ds/Hash.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
+class HashTable {
```
Comment (HarshCasper) [inline JavaScript/ds/Hash.js:25]: Add a beginner-friendly pitch to help the others know what exactly does your code does

File: JavaScript/ds/Hash.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+let keys = myHashTable.keys();
+console.log(keys);
+
+myHashTable.deleteItem("apple");
+keys = myHashTable.keys();
+console.log(keys);
+
+myHashTable.deleteItem("fruits");
+console.log(myHashTable);
```
Comment (HarshCasper) [inline JavaScript/ds/Hash.js]: Make the input dynamic rather than static

Comment (HarshCasper) [inline JavaScript/ds/Hash.js]: Add sample I/O at the end of the Code once you add dynamic input

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: JavaScript/ds/Hash.js (hunk 2 above)
Comment (TidbitsJS) [reply JavaScript/ds/Hash.js]: Can I use `prompt-sync` of the nodejs for dynamic input?

Comment (HarshCasper) [reply JavaScript/ds/Hash.js]: Yes

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by TidbitsJS** [comment]
I have made the changes @HarshCasper 
Let me know if it's good or not

File: JavaScript/ds/Hash.js [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+      value = prompt("Enter Value to delete - ");
+      myHashTable.deleteItem(value);
+      console.log("After Deletion Hash Table - ", myHashTable.keys());
+      break;
+
+    default:
+      console.log("You know this is wrong, Bbye!\n".wrong);
+      return;
+  }
+} while (choice !== 0);
```
Comment (HarshCasper) [inline JavaScript/ds/Hash.js:172]: Add sample I/O at the end of the code

**Review by HarshCasper — CHANGES_REQUESTED** [review]
One tiny change requested: Otherwise good to merge!

File: JavaScript/ds/Hash.js (hunk 3 above)
Comment (TidbitsJS) [reply JavaScript/ds/Hash.js:172]: Added sample I/O
Let me know about further changes @HarshCasper 

P.S. If the I/O scenario that I have created is good, I will like to do it for other cases as well  ( Ex., Linked List in JS )

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**